core/frontend/src/components/autopilot/EndpointCreationDialog.vue

```diff
@@ -0,0 +1,176 @@
+          />
+
+          <v-text-field
+            v-model="endpoint.owner"
+            :counter="50"
+            label="Owner"
+            :rules="[required_rule]"
+          />
+
+          <v-select
+            v-model="endpoint.connection_type"
+            :items="endpoint_types"
+            label="Type"
+            :rules="[required_rule]"
+          />
+
+          <v-text-field
+            v-model="endpoint.place"
+            :rules="[required_rule, ip_path_rule]"
+            label="IP/Device"
+          />
+
+          <v-text-field
+            v-model="endpoint.argument"
+            :counter="50"
+            label="Port/Baudrate"
+            :rules="[required_rule, port_baudrate_rule]"
+          />
+
+          <v-checkbox
+            v-model="endpoint.persistent"
+            label="Save endpoint between system sessions?"
+          />
+
+          <v-checkbox
+            v-model="endpoint.protected"
+            label="Protect endpoint from being deleted?"
+            disabled
+          />
+
+          <v-btn
+            color="success"
+            class="mr-4"
+            @click="createEndpoint"
+          >
+            Create
+          </v-btn>
+        </v-form>
+      </v-card-text>
+    </v-card>
+  </v-dialog>
+</template>
+
+<script lang="ts">
+import axios from 'axios'
+import Vue from 'vue'
+import { getModule } from 'vuex-module-decorators'
+
+import AutopilotStore from '@/store/autopilot'
+import NotificationStore from '@/store/notifications'
+import { EndpointType, userFriendlyEndpointType } from '@/types/autopilot'
+import { autopilot_service } from '@/types/frontend_services'
+import { LiveNotification, NotificationLevel } from '@/types/notifications'
+import { VForm } from '@/types/vuetify'
+import {
+  validate_baudrate, validate_filepath, validate_integer_string, validate_ip, validate_non_empty, validate_port,
+} from '@/utils/pattern_validators'
+
+const notification_store: NotificationStore = getModule(NotificationStore)
+const autopilot_store: AutopilotStore = getModule(AutopilotStore)
+
+export default Vue.extend({
+  name: 'ConnectionDialog',
+  model: {
+    prop: 'show',
+    event: 'change',
+  },
+  props: {
+    show: {
+      type: Boolean,
+      default: false,
+    },
+  },
+
+  data() {
+    return {
+      endpoint: {
+        name: '',
+        owner: '',
+        connection_type: EndpointType.udpin,
+        place: '',
+        argument: '',
+        protected: false,
+        persistent: false,
+      },
+    }
+  },
+  computed: {
+    endpoint_types(): {value: EndpointType, text: string}[] {
+      return Object.entries(EndpointType).map(
+        (type) => ({ value: type[1], text: userFriendlyEndpointType(type[1]) }),
+      )
+    },
+    form(): VForm {
+      return this.$refs.form as VForm
+    },
+  },
+  methods: {
+    required_rule(input: string): (true | string) {
+      return validate_non_empty(input) ? true : 'Required field.'
```

Comment:
This function is not clear, with the return type.
You can have a "required rule true" and you can have a 'Required field' ?
The idea is that when it's `true` you have a valid input and when you have a string you have an invalid input ?
If that's the case, maybe we could rename it to `validate_required_field` ?

Reply:
If that's not the case, we could have what is the actual behavior of this `_rules` as a general comment in methods or anywhere, since this appears to be not obvious looking at the code.

Reply:
This is the Vuetify way to handle validation rules. You have a function that returns `true` (validated) or a `string` (not validated). The `string` should contain the reason why it's not validated.

Reply:
Thanks for the explanation, but the function name does not completely matches what you are saying, I would recommend change it from `required_` to `validate_`, since this is what the Vuetify field does.

Reply:
And I liked the idea of renaming it to `validate_required_field`. Sounds more clear.